### PR TITLE
update release config for 4.4.0

### PR DESCRIPTION
**Pull request by daxmc99**
Finalize https://github.com/sourcegraph/sourcegraph/issues/45566

## Test plan

n/a

## App preview:

- [Web](https://sg-web-dax-finish-release.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: dev/release/release-config.jsonc [hunk 1]
```diff
@@ -11,16 +11,16 @@
...
+    "captainGitHubUsername": "BolajiOlajide",
     // Release versions
-    "previousRelease": "4.2.1",
-    "upcomingRelease": "4.3.0",
-    "oneWorkingWeekBeforeRelease": "8 December 2022 10:00 PST",
-    "threeWorkingDaysBeforeRelease": "12 December 2022 10:00 PST",
-    "releaseDate": "15 December 2022 10:00 PST",
-    "oneWorkingDayAfterRelease": "16 December 2022 10:00 PST",
-    "oneWorkingWeekAfterRelease": "22 December 2022 10:00 PST",
+    "previousRelease": "4.3.0",
```
Comment (BolajiOlajide) [inline dev/release/release-config.jsonc:17]:
```suggestion
    "previousRelease": "4.4.0",
```

File: dev/release/release-config.jsonc [hunk 2]
```diff
@@ -11,16 +11,16 @@
 {
     // Captain information
     // To get your slack username, navigate to Profile -> More -> Account settings -> Username (at the bottom) -> Expand
-    "captainSlackUsername": "dax",
-    "captainGitHubUsername": "daxmc99",
+    "captainSlackUsername": "@Bolaji",
```
Comment (BolajiOlajide) [inline dev/release/release-config.jsonc:14]:
```suggestion
    "captainSlackUsername": "bolaji",
```

**Review by BolajiOlajide — APPROVED** [review]

File: dev/release/release-config.jsonc (hunk 1 above)
Comment (daxmc99) [reply dev/release/release-config.jsonc:17]: Are you sure about this? The release issue says: Change the upcoming release to the current patch release. That would be release that just came out

Comment (BolajiOlajide) [reply dev/release/release-config.jsonc:17]: Dang! I;m sorry please ignore my suggestion. Haven't had enough coffee today.

Comment (BolajiOlajide) [reply dev/release/release-config.jsonc:17]: Dang! I'm sorry, please ignore my suggestion. I haven't had enough coffee today.

File: dev/release/release-config.jsonc [hunk 3]
```diff
@@ -11,16 +11,16 @@
 {
     // Captain information
     // To get your slack username, navigate to Profile -> More -> Account settings -> Username (at the bottom) -> Expand
-    "captainSlackUsername": "dax",
-    "captainGitHubUsername": "daxmc99",
+    "captainSlackUsername": "bolaji",
```
Comment (BolajiOlajide) [inline dev/release/release-config.jsonc:14]:
```suggestion
    "captainSlackUsername": "bolaji.olajide",
```

I should have checked slack before making that suggestion. But here's the correct one,😆.

**Review by BolajiOlajide — APPROVED** [review]

**Pull request merged**